aql-engine/src/main/java/org/ehrbase/openehr/aqlengine/service/AqlQueryServiceImp.java

```diff
@@ -183,34 +183,34 @@ private QueryResultDto queryAql(AqlQueryRequest aqlQueryRequest) {
         }
     }
 
-    private static AqlQuery buildAqlQuery(AqlQueryRequest aqlQueryRequest) {
+    static AqlQuery buildAqlQuery(AqlQueryRequest aqlQueryRequest) {
 
         AqlQuery aqlQuery = AqlQueryParser.parse(aqlQueryRequest.queryString());
 
         // apply limit and offset - where the definitions from the aql are the precedence
-        Long fetch = aqlQueryRequest.fetch();
-        if (fetch != null) {
-            if (aqlQuery.getLimit() != null) {
-                throw new InvalidApiParameterException(
-                        "Invalid AQL query: fetch is defined on query %s and as parameter %s"
-                                .formatted(aqlQuery.getLimit(), fetch));
+        Long fetchParam = aqlQueryRequest.fetch();
+        Long offsetParam = aqlQueryRequest.offset();
+        Long limitQuery = aqlQuery.getLimit();
+
+        // verify not parameter fetch offset are defined when query contains a LIMIT or assign fetch parameter
+        if (limitQuery == null) {
+            aqlQuery.setLimit(fetchParam);
+        } else {
+            if (fetchParam != null || offsetParam != null) {
+                throw new UnprocessableEntityException(
+                        "Query contains a LIMIT clause, fetch and offset parameters must not be used");
             }
-            aqlQuery.setLimit(fetch);
         }
-        Long offset = aqlQueryRequest.offset();
-        if (offset != null) {
-            if (aqlQuery.getOffset() != null) {
-                throw new InvalidApiParameterException(
-                        "Invalid AQL query: fetch is defined on query %s and as parameter %s"
-                                .formatted(aqlQuery.getOffset(), offset));
-            }
-            aqlQuery.setOffset(offset);
+
+        // assign offset parameter
+        if (aqlQuery.getOffset() == null) {
+            aqlQuery.setOffset(offsetParam);
```

Comment:
This can be done (unconditionally) in `if (limitQuery == null)`

Reply:
ups ... slipped in, was from a local test